chirp/drivers/retevis_ha1g.py

```diff
@@ -0,0 +1,2111 @@
+    status = chirp_common.Status()
+    status.msg = "Uploading to radio"
+    status.cur = 0
+    status.max = self._memsize
+    data_bytes = self.get_mmap()
+    for item in MemoryRegions:
+        if (item in [MemoryRegions.radioHead, MemoryRegions.radioInfo,
+                     MemoryRegions.radioVer, MemoryRegions.zoneData]):
+            continue
+        item_bytes = get_write_item_bytes(data_bytes, item)
+        write_item_current_page_bytes(
+            self, serial, item_bytes, item.value, status)
+
+
+def handshake_handle_connect_event(self, dataByte: bytes):
+    if dataByte[14:16] == b"\x00\x01":
+        if dataByte[20] != 1:
+            model_str = self.MODEL
+            radioType = dataByte[20: 20 + len(model_str)]
+            if radioType == model_str.encode("ascii"):
+                return HandshakeStatuses.Normal
+            else:
+                return HandshakeStatuses.RadioWrong
+        else:
+            return HandshakeStatuses.PwdWrong
+    else:
+        return HandshakeStatuses.Wrong
+
+
+def handle_connect_ver(
+    send_byte: bytes, serial_conn, chunk_size=1050, max_retries=5
+) -> tuple[bool, bytes]:
+    serial_conn.write(send_byte)
+    new_bytes = serial_conn.read(chunk_size)
+    if rdtp_page_data_crc16_ver(new_bytes):
+        return True, new_bytes
+    return False, new_bytes
+
+
+def rdtp_page_data_crc16_ver(current_Page_Byte: bytes, chunk_size=1024):
+    byteLen = len(current_Page_Byte)
+    if byteLen <= 13:
+        return False
+    page_count = 255 if chunk_size > 255 else chunk_size
+    pageLen = current_Page_Byte[12] - 6 if current_Page_Byte[12] > 6 else 0
+    pageLen += current_Page_Byte[13] * page_count
+    if byteLen - 23 < pageLen:
+        return False
+    crcBytes = current_Page_Byte[-3:-1]
+    dataBytes = current_Page_Byte[:-3]
+    crcStr = calculate_crc16(dataBytes, 2)
+    return crcBytes == crcStr
+
+
+def get_handshake_bytes(currentModel):
+    """
+    Fully optimized version using struct.pack for all components.
+    """
+    model_bytes = currentModel.encode("ascii")
+    if len(model_bytes) > 41:
+        raise ValueError("Model string too long (max 41 bytes)")
+    magic = b"RDTP\x01\x00\x00\x00\x00\x00\x00\x00"
+    prefix = b"\x00\x00\x00\x00\x01\x00"
+    padding_len = 41 - len(model_bytes)
+    padding = b"\x00" * padding_len
+    data_len = len(prefix) + len(model_bytes) + padding_len
+    packet = struct.pack(
+        f"<12sH6s{len(model_bytes)}s{padding_len}s",
+        magic,
+        data_len,
+        prefix,
+        model_bytes,
+        padding)
+    # Calculate CRC and append footer
+    crc = calculate_crc16(packet, 2)
+    return packet + crc + b"\xff"
+
+
+def get_read_current_page_bytes(self, item: int, serial, status):
+    page_count = 1
+    page_index = 0
+    item_bytes = b""
+    max_retries = 5
+    retry_delay = 0.05
+    while page_index < page_count:
+        success = False
+        for attempt in range(max_retries):
```

Comment:
Why are you retrying here? If we're already talking to the radio and it has responded so far, you should not need to retry in the middle of a communication unless you have some other issue with the protocol. The delay on L643 is not a good sign that you're implementing the protocol properly. Is this perhaps left-over from debugging and can now be removed?